### PR TITLE
Improved the logging

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
Logs no longer overwrite each other, now they append to the log file.

Added starting and ending block to logging to differentiate different upload runs.

### CODE REVIEW

File: core/cli_entry.py [hunk 1]
```diff
@@ -77,4 +79,27 @@ def validate_and_upload_single_entry(directory):
 
     logging.info("Samples in directory '{}' have finished uploading!".format(directory))
 
+    return exit_success()
+
+
+def exit_error():
```
Comment (tom114) [inline core/cli_entry.py]: Comments for all these new methods please.

File: docs/index.md [hunk 2]
```diff
@@ -106,6 +106,17 @@ Use the the `irida-uploader.sh` script included with the source code to upload.
 
 `./irida-uploader.sh /path/to/the/sequencing/run/`
 
+## Logging
+
+Logs are written to your system default logging directory
+
+#### Linux
+
+`/home/<user>/.cache/irida_uploader/log/`
```
Comment (tom114) [inline docs/index.md]: Something i'm noticing is that when I run the tests it logs into this same file.  Any way we can get the tests to dump somewhere else?

Comment (JeffreyThiessen) [reply docs/index.md]: Added a environment variable (that does not persist after tests are run), to write tests logs to a different directory.

Added a note to the readme on where tests log.

File: core/cli_entry.py (hunk 1 above)
Comment (JeffreyThiessen) [reply core/cli_entry.py]: added

File: core/logger.py [hunk 3]
```diff
@@ -1,28 +1,42 @@
 from appdirs import user_log_dir
 import os
-import logging
+import logging.handlers
 
 
 log_directory_name = "irida_uploader"
-log_file_name = 'irida-uploader.log'
+if os.environ.get('IRIDA_UPLOADER_TEST'):
+        log_directory_name = "irida_uploader_test"
```
Comment (tom114) [inline core/logger.py]: Add a comment here please to say what's going on.  Otherwise things look good!

Comment (JeffreyThiessen) [reply core/logger.py]: Added comments explaining each line

**Pull request merged**